chapters/ru/chapter7/4.mdx

```diff
@@ -0,0 +1,1002 @@
+```
+
+{:else}
+
+Чтобы получить из результатов модели тексты, которые может использовать метрика, мы воспользуемся методом `tokenizer.batch_decode()`. Нам просто нужно очистить все значения `-100` в метках (токенизатор автоматически сделает то же самое для дополняющего токена):
+
+```py
+import numpy as np
+
+
+def compute_metrics(eval_preds):
+    preds, labels = eval_preds
+    # В случае, если модель возвращает больше, чем предсказанные логиты
+    if isinstance(preds, tuple):
+        preds = preds[0]
+
+    decoded_preds = tokenizer.batch_decode(preds, skip_special_tokens=True)
+
+    # Заменяем -100 в метках, так как мы не можем их декодировать
+    labels = np.where(labels != -100, labels, tokenizer.pad_token_id)
+    decoded_labels = tokenizer.batch_decode(labels, skip_special_tokens=True)
+
+    # Немного простой постобработки
+    decoded_preds = [pred.strip() for pred in decoded_preds]
+    decoded_labels = [[label.strip()] for label in decoded_labels]
+
+    result = metric.compute(predictions=decoded_preds, references=decoded_labels)
+    return {"bleu": result["score"]}
+```
+
+{/if}
+
+Теперь, когда все готово, мы готовы дообучить нашу модель!
+
+
+### Дообучение модели[[fine-tuning-the-model]]
+
+Первый шаг - войти в Hugging Face, чтобы загрузить результаты в Model Hub. В блокноте есть удобная функция, которая поможет вам в этом:
+
+```python
+from huggingface_hub import notebook_login
+
+notebook_login()
+```
+
+Появится виджет, в котором вы можете ввести свои учетные данные для входа в Hugging Face.
+
+Если вы работаете не в блокноте, просто введите следующую строку в терминале:
+
+```bash
+huggingface-cli login
+```
+
+{#if fw === 'tf'}
+
+Прежде чем начать, давайте посмотрим, какие результаты мы получим от нашей модели без какого-либо обучения:
+
+```py
+print(compute_metrics())
+```
+
+```
+{'bleu': 33.26983701454733}
+```
+
+Как только это будет сделано, мы сможем подготовить все необходимое для компиляции и обучения нашей модели. Обратите внимание на использование `tf.keras.mixed_precision.set_global_policy("mixed_float16")` - это укажет Keras обучать с использованием float16, что может дать значительное ускорение на GPU, поддерживающих эту функцию (Nvidia 20xx/V100 или новее).
+
+```python
+from transformers import create_optimizer
+from transformers.keras_callbacks import PushToHubCallback
+import tensorflow as tf
+
+# Количество шагов обучения - это количество примеров в датасете, разделенное на размер батча, затем умноженное
+# на общее количество эпох. Обратите внимание, что tf_train_dataset здесь - это батч tf.data.Dataset,
+# а не оригинальный датасет Hugging Face, поэтому его len() уже равен num_samples // batch_size.
+num_epochs = 3
+num_train_steps = len(tf_train_dataset) * num_epochs
+
+optimizer, schedule = create_optimizer(
+    init_lr=5e-5,
+    num_warmup_steps=0,
+    num_train_steps=num_train_steps,
+    weight_decay_rate=0.01,
+)
+model.compile(optimizer=optimizer)
+
+# Train in mixed-precision float16
```

Comment:
Please translate the comment

Reply:
Done in 5eb4f4be1320f29454758cc7def43e1bb2e4f1a0